packages/api-server/src/convert-tx.ts

```diff
@@ -35,6 +44,35 @@ export interface PolyjuiceTransaction {
   s: HexString;
 }
 
+export function polyjuiceRawTransactionToApiTransaction(
+  rawTx: HexString,
+  ethTxHash: Hash,
+  tipBlockHash: Hash,
+  tipBlockNumber: bigint,
+  fromEthAddress: HexString
+): EthTransaction {
+  const tx: PolyjuiceTransaction = decodeRawTransactionData(rawTx);
+
+  const pendingBlockHash = bumpHash(tipBlockHash);
+  const pendingBlockNumber = new Uint64(tipBlockNumber + 1n).toHex();
+  return {
+    hash: ethTxHash,
+    blockHash: pendingBlockHash,
+    blockNumber: pendingBlockNumber,
+    transactionIndex: "0x0",
```

Comment:
It can reuse [`PENDING_TRANSACTION_INDEX`](https://github.com/nervosnetwork/godwoken-web3/blob/4d3dddc5881fedf0dbb8cce8a250aa423f9f5e99/packages/api-server/src/filter-web3-tx.ts#L29): `transactionIndex: PENDING_TRANSACTION_INDEX,`

Reply:
Updated